src/LineReader.php

```diff
@@ -1,17 +1,21 @@
 <?php
 
+declare(strict_types=1);
+
 namespace Bcremer\LineReader;
 
 final class LineReader
 {
     /**
      * Prevent instantiation
      */
-    private function __construct() {}
+    private function __construct()
+    {
+    }
 
     /**
-     * @param string $filePath
-     * @return \Generator
+     * @return \Generator<string>
```

Comment:
If we're going to introduce generic types we can also define the full type as  `\Generator<int, string, void, void>`.

This will give further type grantees about the genrator.
See
- https://phpstan.org/r/e20abfae-19df-4b64-85bd-006368649a1c
- https://psalm.dev/r/c1af9c0927

Reply:
TIL :grin: 

thanks :+1: